junit-platform-launcher/src/main/java/org/junit/platform/launcher/TestIdentifier.java

```diff
@@ -90,19 +78,11 @@ private TestIdentifier(UniqueId uniqueId, String displayName, @Nullable TestSour
 		this.parentId = parentId;
 		this.displayName = displayName;
 		this.source = source;
-		this.tags = copyOf(tags);
+		this.tags = new LinkedHashSet<>(tags);
```

Comment:
Please revert this particular change and restore the `copyOf()` method. Since most descriptors don't have tags it's a non-negligible optimization.